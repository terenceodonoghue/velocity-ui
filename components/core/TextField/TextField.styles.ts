import { css, FunctionInterpolation, Theme } from '@emotion/react';
import { darken, em, rgba, transitions as transition } from 'polished';

export const textFieldInput: FunctionInterpolation<Theme> = ({
  palette,
  transitions,
  typography,
}) => css`
  display: block;
  border: solid 1px ${palette.secondary};
  border-radius: 5px;
  height: 38px;
  margin: 7px 0;
  max-width: 270px;
  padding: 10px 16px;
  width: 100%;
  background-color: ${rgba(palette.secondary, 0.2)};
  color: ${palette.neutral[900]};
  font-family: inherit;
  font-size: ${em(15, typography.fontSize)};
  ${transition(
    ['border', 'opacity'],
    `${transitions.duration.standard}ms ${transitions.easing.sharp}`,
  )};

  &:disabled {
    opacity: 0.3;
  }

  &:focus {
    border: solid 1px ${palette.accent};
    outline: 0;
  }
`;

export const textFieldLabel: FunctionInterpolation<Theme> = ({
  palette,
  typography,
}) => css`
  display: inline-block;
  color: ${darken(palette.tonalOffset, palette.neutral[600])};
  font-size: 12px;
  font-weight: ${typography.fontWeight.medium};
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;
