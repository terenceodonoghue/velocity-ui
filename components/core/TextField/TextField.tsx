import React, { forwardRef, InputHTMLAttributes } from 'react';
import * as css from './TextField.styles';

const TextField = forwardRef<
  HTMLInputElement,
  InputHTMLAttributes<HTMLInputElement>
>(({ children, ...props }, ref) => (
  <label css={css.textFieldLabel} data-testid="textfield">
    {children}
    <input
      css={css.textFieldInput}
      data-testid="textfield-input"
      ref={ref}
      {...props}
    />
  </label>
));

export default TextField;
