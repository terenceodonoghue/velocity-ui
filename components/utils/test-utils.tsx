import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
