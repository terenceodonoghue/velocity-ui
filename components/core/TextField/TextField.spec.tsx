import React from 'react';
import faker from 'faker';
import { render, userEvent } from 'test-utils';
import TextField from './TextField';

describe('TextField', () => {
  it('renders label', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId } = render(<TextField>{content}</TextField>);

    expect(getByTestId('textfield')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });

  it('is empty by default', () => {
    const { getByTestId } = render(<TextField />);

    expect(getByTestId('textfield-input')).not.toHaveValue();
  });

  it('shows input when typing', () => {
    const content = faker.hacker.phrase();
    const { getByTestId } = render(<TextField />);

    userEvent.type(getByTestId('textfield'), content);
    expect(getByTestId('textfield-input')).toHaveValue(content);
  });
});
