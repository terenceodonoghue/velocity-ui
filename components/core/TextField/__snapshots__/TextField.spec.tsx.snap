// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders label 1`] = `
.emotion-0 {
  display: inline-block;
  color: #54657a;
  font-size: 12px;
  font-weight: 500;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

.emotion-1 {
  display: block;
  border: solid 1px #e0e7ff;
  border-radius: 5px;
  height: 38px;
  margin: 7px 0;
  max-width: 270px;
  padding: 10px 16px;
  width: 100%;
  background-color: rgba(224,231,255,0.2);
  color: #2E384D;
  font-family: inherit;
  font-size: 1em;
  -webkit-transition: border 300ms cubic-bezier(0.4, 0, 0.6, 1),opacity 300ms cubic-bezier(0.4, 0, 0.6, 1);
  transition: border 300ms cubic-bezier(0.4, 0, 0.6, 1),opacity 300ms cubic-bezier(0.4, 0, 0.6, 1);
}

.emotion-1:disabled {
  opacity: 0.3;
}

.emotion-1:focus {
  border: solid 1px #2e5bff;
  outline: 0;
}

<div>
  <label
    class="emotion-0"
    data-testid="textfield"
  >
    The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
    <input
      class="emotion-1"
      data-testid="textfield-input"
    />
  </label>
</div>
`;
